src/main/java/com/hotels/beeju/HiveServer2JUnitRule.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (C) 2015-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.UUID;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hive.jdbc.HiveDriver;
+import org.apache.hive.service.server.HiveServer2;
+import org.apache.thrift.TException;
+import org.hsqldb.jdbc.JDBCDriver;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.hotels.beeju.server2.RelaxedSQLStdHiveAuthorizerFactory;
+
+/**
+ * A JUnit Rule that creates a Hive Server 2 service and Thrift service backed by a Hive Metastore using an HSQLDB
+ * in-memory database.
+ * <p>
+ * A fresh database instance will be created for each test method.
+ * <p>
+ * Note that this class is meant to be used for DDL operations only. Any attempt to run DML operations may results in
+ * errors.
+ */
+public class HiveServer2JUnitRule extends ExternalResource {
+
+  public static final String HSQLDB_USER = "user";
+  public static final String HSQLDB_PASSWORD = "password";
+
+  @VisibleForTesting
+  final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  protected final HiveConf conf = new HiveConf();
+  private final String databaseName;
+  private final String metastoreConnectionURL;
+  private final String driverClassName;
+  private String connectionUrl;
+  private HiveServer2 hiveServer2;
+  private File metastoreLocation;
+  private int port;
+
+  /**
+   * Create a Hive Server 2 service with a pre-created database "test_database".
+   */
+  public HiveServer2JUnitRule() {
+    this("test_database");
+  }
+
+  /**
+   * Create a Hive Server 2 service with a pre-created database using the provided name.
+   *
+   * @param databaseName Database name.
+   */
+  public HiveServer2JUnitRule(String databaseName) {
+    checkNotNull(databaseName, "databaseName is required");
```

Comment:
That, although it might get a bit tricky to do all of it since this modifies quite a bit of internal object state. Another option would be to introduce an inheritance hierarchy or use composition with a BeejuConfiguration object which holds all that state and initialises it to those default values. My vote goes for the latter option.